OpenEdX/Managers/SegmentAnalyticsManager/SegmentAnalyticsManager.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  SegmentAnalyticsManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 21/02/2024.
+//
+
+import Foundation
+import Core
+import Segment
+import SegmentFirebase
+
+class SegmentAnalyticsManager: AnalyticsService {
+    var analytics: Analytics?
+    
+    // Init manager
+    public init(config: ConfigProtocol) {
+        guard config.segment.enabled else { return }
+
+        let configuration = Configuration(writeKey: config.segment.writeKey)
+                        .trackApplicationLifecycleEvents(true)
+                        .flushInterval(10)
+        analytics = Analytics(configuration: configuration)
+        if config.firebase.isAnalyticsSourceSegment {
```

Comment:
Oh it got missed, please update condition for `config.firebase.enabled` as well so in case of missing keys app will not crash.

Reply:
done